modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ModuleFlowProcessingTemplate.java

```diff
@@ -88,12 +88,10 @@ public void sendAfterTerminateResponseToClient(Either<Event, MessagingException>
                      .onTerminate(right(messagingException)));
   }
 
-  private void runAndNotify(CheckedRunnable runnable, Event event, ResponseCompletionCallback responseCompletionCallback) {
-    try {
-      runnable.run();
-      responseCompletionCallback.responseSentSuccessfully();
-    } catch (Exception e) {
-      responseCompletionCallback.responseSentWithFailure(new MessagingException(event, e), event);
-    }
+  private Mono<Void> runAndNotify(CheckedSupplier<Publisher<Void>> supplier, Event event,
```

Comment:
For reuse like this, I find this approach to be clearer, it's also the same as we are doing everywhere else:
```
  @Override
  public Publisher<Void> sendResponseToClient(Event event, Map<String, Object> parameters,
                                              Function<Event, Map<String, Object>> errorResponseParametersFunction,
                                              ResponseCompletionCallback responseCompletionCallback) {
    return from(completionHandler.onCompletion(event, parameters)).transform(notifyCompletion(event, responseCompletionCallback));
  }

  private Function<Publisher<Void>, Publisher<Void>> notifyCompletion(Event event,
                                                                      ResponseCompletionCallback responseCompletionCallback) {
    return publisher -> from(publisher)
        .doOnSuccess(v -> responseCompletionCallback.responseSentSuccessfully())
        .doOnError(e -> responseCompletionCallback.responseSentWithFailure(new MessagingException(event, e), event));
  }
```